status/flowcell.py

```diff
@@ -109,3 +114,255 @@ def get(self, flowcell_id):
                                   thresholds=thresholds,
                                   project_names=project_names,
                                   user=self.get_current_user()))
+
+
+class ONTReportHandler(SafeHandler):
+    """ Serves a page showing the MinKNOW .html report of a given run
+    """
+
+    def __init__(self, application, request, **kwargs):
+        super(SafeHandler, self).__init__(application, request, **kwargs)
+
+    def fetch_ont_report(self, name):
+        """ Fetches the MinKNOW .html run report, saved as an escaped string in the
+        database run entry and returns the unescaped string which can be written as .html
+        """
+
+        view_report = self.application.nanopore_runs_db.view("info/minknow_report", descending=True)
+        row = [row for row in view_report.rows if name in row.key][0]
+        str_html = html.unescape(row.value)
+
+        return str_html
+
+    def get(self, name):
+        self.write(self.fetch_ont_report(name))
+
+
+class ONTFlowcellHandler(SafeHandler):
+    """ Serves a page which shows information for a given ONT flowcell.
+    """
+
+    def __init__(self, application, request, **kwargs):
+        super(SafeHandler, self).__init__(application, request, **kwargs)
+
+    def fetch_ont_flowcell(self, name):
+
+        view_all = self.application.nanopore_runs_db.view("info/all_stats", descending=True)
+        view_status = self.application.nanopore_runs_db.view("info/run_status", descending=True)
+        view_project = self.application.projects_db.view("project/id_name_dates", descending=True)
+
+        fc = {}
+
+        row = [row for row in view_status.rows if name in row.key][0]
+
+        if row.value == "ongoing":
+            fc["TACA_run_path"] = row.key
+            fc["TACA_run_status"] = row.value
+
+        elif row.value == "finished":
+            name = row.key.split("/")[-1]
+
+            row = view_all[name].rows[0]
+            fc["name"] = row.key
+
+            d = row.value
```

Comment:
I think assigning row.value to d here reduces readability a bit. Better to use row.value, its not that long anyway